pkgs/development/compilers/llvm/17/mlir/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib, stdenv, llvm_meta
+, buildLlvmTools
+, monorepoSrc, runCommand
+, cmake
+, ninja
+, libxml2
+, libllvm
+, version
+, doCheck ? !stdenv.hostPlatform.isMusl
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mlir";
+  inherit version doCheck;
+
+  # Blank llvm dir just so relative path works
+  src = runCommand "${pname}-src-${version}" {} ''
+    mkdir -p "$out"
+    cp -r ${monorepoSrc}/cmake "$out"
+    cp -r ${monorepoSrc}/${pname} "$out"
+    cp -r ${monorepoSrc}/third-party "$out/third-party"
+    mkdir -p "$out/llvm"
+  '';
+
+  sourceRoot = "${src.name}/${pname}";
+
+  patches = [
+    ./gnu-install-dirs.patch
+  ];
+
+  nativeBuildInputs = [ cmake ninja ];
+  buildInputs = [ libllvm libxml2 ];
+
+  ninjaFlags = [ "-v " ];
+  cmakeFlags = [
+    "-DLLVM_BUILD_TOOLS=ON"
```

Comment:
lib.cmake* would be interesting

Reply:
I didn't get this? could you please elaborate?

Reply:
I think they mean `(lib.cmakeBool "LLVM_BUILD_TOOLS" true)` instead of `"-DLLVM_BUILD_TOOLS=ON"`.

Reply:
ahhh interesting, but just to follow the same style as other modules in the llvm dir let's keey it this way